pkg/plugin/step.go

```diff
@@ -74,13 +74,17 @@ func parseIntervalStringToTimeDuration(interval string) (time.Duration, error) {
 }
 
 // calculateStep calculates step by provided max datapoints and timerange
-func calculateStep(minInterval time.Duration, from, to time.Time, maxDataPoints int64) time.Duration {
-	resolution := maxDataPoints
+func (q *Query) calculateStep(minInterval time.Duration) time.Duration {
+	if q.Instant && q.Interval == "" {
+		return instantQueryDefaultStep
+	}
```

Comment:
The condition is not completely correct.
I assume, that `q.Interval == ""` when user omitted it (set to default) and `q.Interval != ""` when user specified it explicitly. 

The logic then should be the following:
```
	if  q.Interval == "" {
// do what user wants
		return  q.Interval
	}
// if user set to auto, let's do custom logic
if q.Instant {
  return instantQueryDefaultStep
}
// if not instant, let's auto-calculate
resolution := q.MaxDataPoints
....
```